components/base-war/src/main/webapp/templates/macros.vm

```diff
@@ -2742,3 +2742,20 @@ Recursive title display detected!##
   #set ($sentence3 = 'Click this box to get technical details.')
   #displayException("${sentence1} ${sentence2} ${sentence3}", $exception)
 #end
+
+###
+## Display external ID page title in case it is configured to be shown instead of internal ID
+###
+### @param defaultTitle The default page title to format for displaying
+#macro(pageExternalTitle $defaultTitle)
+#set($config = $xwiki.getDocument('data.WebHome').getObject('PhenoTips.DBConfigurationClass'))
+#set($pageExternalTitle = $defaultTitle)
+#if($config && "$!{config.getProperty('hideInternalIds').value}" == '1' && "$!{doc.getValue('external_id')}" != '')
+  #set($ext_id = "$!{doc.getValue('external_id')}")
+  #if("$!{config.getProperty('externalIdFormat').value}" == 'XXX-XXX-XXX')
+    #set($ext_id = $ext_id.replaceAll("(.{3})(?!$)", "$1-"))
```

Comment:
Use this instead:

    $ext_id.replaceAll("([^-])(?=([^-]{3})+(?=(-|$)))", "$1-")

Replace any non-dash character that is followed by 3 other non-dash characters then by either dash or end with that character and a dash.